seahorn/lib/hash_table_helper.c

```diff
@@ -0,0 +1,35 @@
+#include <aws/common/math.h>
+#include <hash_table_helper.h>
+#include <proof_allocators.h>
+#include <seahorn/seahorn.h>
+#include <utils.h>
+
+void initialize_bounded_aws_hash_table(struct aws_hash_table *map,
+                                       size_t max_table_entries) {
+  size_t num_entries = nd_size_t();
+  assume(num_entries <= max_table_entries);
+  assume(aws_is_power_of_two(num_entries));
+
+  size_t required_bytes = nd_size_t();
+  /* assume setting required_bytes is successful */
+  assume(!hash_table_state_required_bytes(num_entries, &required_bytes));
+  struct hash_table_state *impl = bounded_malloc(required_bytes);
+  impl->size = num_entries;
+  map->p_impl = impl;
+}
+
+void save_byte_from_hash_table(const struct aws_hash_table *map,
+                               struct store_byte_from_buffer *storage) {
+  struct hash_table_state *state = map->p_impl;
+  size_t size_in_bytes = nd_size_t();
+  assume(hash_table_state_required_bytes(state->size, &size_in_bytes) ==
+         AWS_OP_SUCCESS);
+  save_byte_from_array((uint8_t *)state, size_in_bytes, storage);
+}
+
+void check_hash_table_unchanged(const struct aws_hash_table *map,
```

Comment:
`assert_hash_table_unchanged` to be consistent. But maybe this is how it is called in CBMC.
I would still change the name